pkg/frontend/admin_ovsflows_get.go

```diff
@@ -0,0 +1,195 @@
+	for i := 0; pod.Status.Phase != corev1.PodSucceeded; i++ {
+		if i == 60 {
+			err = api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeInternalServerError, "", "")
+			return nil, err
+		}
+		pod, err = config.CoreV1().Pods(OVSFlowPodNamespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Error(err)
+			return nil, err
+		}
+		time.Sleep(time.Second)
+	}
+
+	// The OVS flows would be present in the logs of the created pod.
+	b, err := client.getPodLogs(ctx, log, pod)
+	if err != nil {
+		return nil, err
+	}
+
+	// Clearing the pod after extracting its logs.
+	defer func() {
+		config.CoreV1().Pods(OVSFlowPodNamespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+
+	return b, nil
+}
+
+type Client struct {
+	Clientset kubernetes.Interface
+}
+
+// This Function extract logs from the passed pod and return in the form of a slice bytes.
+func (c Client) getPodLogs(ctx context.Context, log *logrus.Entry, pod *corev1.Pod) ([]byte, error) {
+	opts := corev1.PodLogOptions{Container: OVSFlowContianerName}
+	r := c.Clientset.CoreV1().Pods(OVSFlowPodNamespace).GetLogs(pod.Name, &opts)
+	result, err := r.Do(ctx).Raw()
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+	return result, err
+}
+
+func (c Client) createOVSFlowPod(ctx context.Context, node string, log *logrus.Entry) (*corev1.Pod, error) {
+	var hostpathType corev1.HostPathType = "Directory"
+	hostPath := corev1.VolumeSource{HostPath: &corev1.HostPathVolumeSource{"/", &hostpathType}}
+	volumes := []corev1.Volume{{"host", hostPath}}
+	privileged := true
+	var runAsUser int64 = 0
+	i := 0
+	podName := OVSFlowPodName
+	for {
+		pod := &corev1.Pod{
+			metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: OVSFlowPodNamespace,
+			},
+			corev1.PodSpec{
+				Volumes: volumes,
+				Containers: []corev1.Container{{
+					Name:    OVSFlowContianerName,
+					Image:   "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:103505c93bf45c4a29301f282f1ff046e35b63bceaf4df1cca2e631039289da2",
+					Command: []string{"chroot", "/host", "/bin/bash", "-c", "ovs-ofctl -O OpenFlow13 dump-flows br0"},
+					VolumeMounts: []corev1.VolumeMount{{
+						Name:      "host",
+						MountPath: "/host",
+					}},
+					SecurityContext: &corev1.SecurityContext{
+						Privileged: &privileged,
+						RunAsUser:  &runAsUser,
+					},
+				}},
+				RestartPolicy: "Never",
+				NodeName:      node,
+				HostNetwork:   true,
+				HostPID:       true,
+			},
+			corev1.PodStatus{},
+		}
+		pod, err := c.Clientset.CoreV1().Pods("default").Create(ctx, pod, metav1.CreateOptions{})
+		if err != nil && err.Error() == fmt.Sprintf("pods \"%s\" already exists", podName) {
+			i++
+			podName = fmt.Sprintf("%s-%d", OVSFlowPodName, i)
+			continue
+		} else if err != nil {
+			log.Error(err)
+			return nil, err
+		}
```

Comment:
I would like to have a static pod name. Maybe delete before creating instead?

Reply:
Ohh .. so you mean delete the existing pod and then create the pod with the same name .. if the pod already exists ?
Actually .. I thought in a way where someone is running the pod in the default project with the same name as our pod .. then we would not like to touch it .. we just need to do our work and go..